packages/babel-plugin-transform-es2015-duplicate-keys/src/index.js

```diff
@@ -0,0 +1,38 @@
+import * as t from "babel-types";
+
+function getName(key) {
+  if (t.isIdentifier(key)) {
+    return key.name;
+  }
+  return key.value.toString();
+}
+
+export default function() {
+  return {
+    visitor: {
+      ObjectExpression(path) {
+        const { node } = path;
+        const plainProps = node
+          .properties
+          .filter(prop => !t.isSpreadProperty(prop) && !prop.computed);
+
+        const alreadySeenNames = {};
+
+        plainProps
+          .filter(prop => {
```

Comment:
This `.filter().forEach` is kind of confusing. Can you just loop over `plainProps` with a `for...of` and add an `if` that does the `prop.computed` logic?